clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/ChainResolver.java

```diff
@@ -44,24 +53,50 @@ public ChainResolver(final OperationsCodec<K, V> codec) {
    * @param key a key
    * @return an entry with the resolved operation for the provided key as the key
    * and the compacted chain as the value
-   * @throws ClassNotFoundException
    */
-  public ResolvedChain<K, V> resolve(Chain chain, K key) {
+  public ResolvedChain<K, V> resolve(Chain chain, K key, long now) {
     Result<V> result = null;
     ChainBuilder chainBuilder = new ChainBuilder();
+    Operation<K, V> operation = null;
+    boolean isNew = false;
     for (Element element : chain) {
       ByteBuffer payload = element.getPayload();
-      Operation<K, V> operation = codec.decode(payload);
+      operation = codec.decode(payload);
+      final Result<V> previousResult = result;
+      Duration expiration;
       if(key.equals(operation.getKey())) {
         result = operation.apply(result);
+        if(result == null) {
+          continue;
+        }
+        if((previousResult == null & operation.isFirst())) {
+          expiration = expiry.getExpiryForCreation(key, result.getValue());
+          isNew = true;
+        } else {
+          expiration = expiry.getExpiryForUpdate(key, new ValueSupplier<V>() {
+            @Override
+            public V value() {
+              return previousResult != null ? previousResult.getValue() : null;
```

Comment:
How could previousResult be null if you are doing an expiry for update check?

Returning null is not legal per the Expiry contract.